cmake/UseCcache.cmake

```diff
@@ -0,0 +1,56 @@
+# Compiling ccache via ccache is fine, because this uses a stable version which
+# is installed on the system.
+
+# Modified version of Craig Scott's "Professional CMake: A Practical Guide", 7th Edition
+function(use_ccache)
+    if(NOT CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+        message(INFO " use_ccache() disabled, as not called at root of project")
+        return()
+    endif()
+
+    find_program(CCACHE_PROGRAM ccache)
+    if(NOT CCACHE_PROGRAM)
+        message(INFO " use_ccache() disabled, as ccache program not found")
+        return()
+    endif()
+
+    message(INFO " use_ccache() enabled")
+
+    # This will override any user settings
+    set(ccacheEnv
+            # Another option would be CMAKE_BINARY_DIR, however currently only one basedir is supported.
+            CCACHE_BASEDIR=${CMAKE_SOURCE_DIR}
+
+            # In case of very old ccache versions (pre 3.3)
+            CCACHE_CPP2=true
+
+            # In case of old ccache versions (pre 4.0)
+            CCACHE_COMPRESS=1
```

Comment:
I think that it would be better to leave this choice to the user. I have personally never used compression with ccache before 4.0.

Reply:
I've been using ccache 3.4.3 with compression for many years now. It works fine :-)